components/brave_shields/browser/ad_block_filters_provider_manager.cc

```diff
@@ -46,7 +46,7 @@ void AdBlockFiltersProviderManager::AddProvider(
   auto it =
       std::find(filters_providers_.begin(), filters_providers_.end(), provider);
   CHECK(it == filters_providers_.end());
```

Comment:
`Use CHECK() if the consequence of a failed assertion would be a security vulnerability, where crashing the browser is preferable.`

https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md#check_dcheck_and-notreached

i suggest replacing most CHECKs with DCHECKs

Reply:
i.e.
```
auto rv = filters_providers_.insert(provider);
DCHECK(rv.second)
```

Reply:
same in `RemoveProvider` below